### DISCUSSION TITLE
Successfully compiling on Mac OS Arm.

### DISCUSSION THREAD

**Discussion by aleph1** [Kode/Kore#583] [Ideas]
I just tried compiling my application on a new MBP with the M1 chip, and it only required a single modification to get it to compile and run as expected. In https://github.com/Kode/Kinc/blob/master/Sources/kinc/simd/float32x4.h I had to move  defined(KORE_MACOS) down to be part of the same if statement as #if defined(ARM64) || defined(KORE_SWITCH). However this change would break compilation on x86 Macs.

Since Mac OS can be on either processor now, would you consider the following?

- KORE_MACOS (denotes all Macs, irrespective of proessor)
- KORE_MACOS_X86 (denotes a mac running on x86)
- KORE_MACOS_ARM (denotes a mac running on ARM)

This wouldn't require many changes to the code base, as it is really only the aforementioned https://github.com/Kode/Kinc/blob/master/Sources/kinc/simd/float32x4.h file that needs both KORE_MACOS_X86 vs KORE_MACOS_ARM, however, it might be useful having those constants for the future.

**Comment by RobDangerous** [comment]
I don't think we need should have a define like that - let's use whatever Xcode itself defines for x64 or ARM instead.
